File: C/Binary_Search.c
#include <stdio.h>

int binary(int a[], int n, int call, int first, int last, int middle)
{
    first = 0;
    last = n - 1;
    middle = (first + last) / 2;
    while (first <= last)
    {
        if (a[middle] < call)
            first = middle + 1;
        else if (a[middle] == call)
        {
            printf("Element %d is found at location %d  \n", call, (middle + 1));
            break;
        }
        else
            last = middle - 1;
        middle = (first + last) / 2;
    }
    if (first > last)
        printf("%d is not present in the array \n", call);
}

int main()
{
    int a[100], n, i, call, first, last, middle;
    printf("Binary search is executed\n\n");
    printf("Enter the number of elements you wish to enter\n");
    scanf("%d", &n); //Elements should be sorted
    printf("Enter the elements in a sorted order for binary search\n\n");
    for (i = 0; i < n; i++) //Scanning Elements
    {
        scanf("%d", &a[i]);
    }
    printf("Enter the element to search\n");
    scanf("%d", &call);                      //Search element
    binary(a, n, call, first, last, middle); //Call to function
    return 0;
}

 /*
    Problem name :- Binary Search
    
    Input:
    Binary search is executed

    Enter the number of elements you wish to enter
    5
    Enter the elements in a sorted order for binary search

    1
    5 
    12  
    16
    19
    Enter the element to search
    16
    
    Output:
    
    Element 16 is found at location 4  
    
    Time Complexity:- O(log n)
    Space Complexity:- O(n)
    */

File: PHP/binary_search.php
<?php
/*
*Binary Search is a searching technique used to search an element in a Sorted array.
*Binary search works with only sorted array.

* Sample I/O
Example1:
 Input: key=10
 Output: 10 key is Exist.

Example2:
 Input: key=17
 Output: 17 key Does not Exist.

 Time Complexity of binary search : O(log n)
 Space Complexity of binary search : O(1)

*/

function binarySearch($array, $key) {
  // check for an empty array
  if (count($array) === 0) return false;
  $low = 0;
  $high = count($array) - 1;

  while ($low <= $high) {
    // compute middle index
    $midpoint = floor(($low + $high) / 2);

    if ($key < $array[$midpoint]) {
      // The key value is less than the midpoint value.
      $high = $midpoint - 1;
    } elseif ($key > $array[$midpoint]) {
      // The key value is greater than the midpoint value.
      $low = $midpoint + 1;
    } else {
        // This is the key we are looking for.
        return $midpoint;
    }
  }
  // The key was not found.
  return false;
}

// Sorted array
$array = array(1, 2, 4, 10, 15, 18, 22);
$key = 4;
if(binarySearch($array, $key) == true) {
    echo $key." key is Exist.";
}
else {
    echo $key." key Does not Exist.";
}
?>
